store/manager.go

```diff
@@ -129,6 +129,8 @@ func (m *Manager) Get(ctx context.Context, key []byte, cm commitments.Commitment
 			m.log.Warn("Failed to read from cache targets", "err", err)
 		}
 
+		ctx = context.WithValue(ctx, common.EncodingCtxKey, cm.Encoding)
```

Comment:
why use context key/value dict everywhere instead of passing as a proper (typed!) argument?

Reply:
this is a workaround to conform with existing interface constraints

Reply:
Why do we need to conform to existing interfaces?

Reply:
The current context-based solution is orthogonal to the correct parameter-based solution, IMO.

I don't think we should introduce the tech debt of context based parameters for the sake of maintaining these internal interfaces.